### PR TITLE
Make a SOGo panel in the server-manager

### PR DESCRIPTION
**Pull request by stephdl**
https://github.com/NethServer/dev/issues/5575

### CODE REVIEW

**Review by DavidePrincipi — COMMENTED** [review]
Remember to add a license header to script files

**Review by markVnl — COMMENTED** [review]
As said before, can't really comment on the nethgui parts..

If we take the clean up of zzz_SOGo.conf on, we might as well clean default-virtualhost.inc up a little bit.
https://github.com/NethServer/nethserver-sogo/blob/master/root/etc/e-smith/templates/etc/httpd/conf.d/default-virtualhost.inc/30SOGo#L17
Although this has nothing to do with the implenetation of the SOGo pannel

File: root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Sogo.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+$L['Dav_label'] = 'Enable CalDAV and CardDAV';
+$L['MailAuxiliaryUserAccountsEnabled_label'] = 'Allow Users to add other IMAP accounts';
+$L['ActiveSync_label'] = 'Enable Microsoft ActiveSync';
+$L['AdminUsers_label'] = 'Administrators';
+$L['Advanced_label'] = 'Advanced settings';
+$L['VirtualHost_label'] = 'Make SOGo reachable only from this domain name list';
+$L['Notifications_label'] = 'Notifications';
+$L['ACLs_label'] = 'Sent by emails to involved users of a calendar/address book';
+$L['Folders_label'] = 'Sent by emails on the creation of a calendar/address book.';
+$L['Appointment_label'] = 'Sent by emails to meeting participants';
```
Comment (markVnl) [inline root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Sogo.php]: I think 
`Send by emails to involved users of a calendar/address book`
can be better phrased:
`Send email to involved users of a calendar/address book `
`Send email on the creation of a calendar/address book`
`Send email to meeting participants`

File: root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Sogo.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+$L['EMail_label'] = "Sent by emails to user's events and tasks";
```
Comment (markVnl) [inline root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Sogo.php]: Can not find this in the SOGo manual,
and also rephrasing as above does not make clear when this notification is sent..
`Send email to user's events and tasks`

File: root/etc/e-smith/templates/etc/httpd/conf.d/zzz_SOGo.conf/10base [hunk 3]
```diff
@@ -1,88 +1,92 @@
...
-                $sogo_cert = 'Sogo';
-                if ( -e '/etc/pki/tls/certs/Sogo-chain.crt'){
-                   $chain_file = "SSLCertificateChainFile \"/etc/pki/tls/certs/Sogo-chain.crt\"\n";
-                }
-            } else {
-                $sogo_cert = 'localhost';
+my $status = $sogod{'status'} || 'disabled';
+return "#SOGo is disabled\n" unless ($status  eq 'enabled');
+
+my @sogo_hosts = split (',', ($sogod{'VirtualHost'} || 'Default'));
```
Comment (markVnl) [inline root/etc/e-smith/templates/etc/httpd/conf.d/zzz_SOGo.conf/10base]: Are we (re)introducing multiple vhost?
If so, 
How to handle the certificates for those hosts? 
Actually the reason it was changed to one single option, 
although that was before let's encrypt was fully developed.

Comment (DavidePrincipi) [reply root/etc/e-smith/templates/etc/httpd/conf.d/zzz_SOGo.conf/10base]: What is the use case of multiple vhost? Is it really needed? 
Can we make it simpler?

Comment (stephdl) [reply root/etc/e-smith/templates/etc/httpd/conf.d/zzz_SOGo.conf/10base]: >What is the use case of multiple vhost? Is it really needed?
Can we make it simpler?

the documentation stated that multiple vhost is possible, I had two solutions, modify the documentation to say only one vhost is usable, this is the simplest way.
Or rewrite the template, but indeed the certificate is probably a non used option, hence I decided to not expose it in the UI

one pro to get multiple vhost is for activesync settings. Imagine you are configuring all settings in you mobile, if an exclusive vhost is given, you think first to mail.subdomain.com, but for a web application it is probably a non sense, it will probably sogo.subdomain.com and you will fail to configure your mobile activesync settings. It is a real case I got it this weekend after I did a reset to my cell

File: root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Sogo.php (hunk 2 above)
Comment (stephdl) [reply root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Sogo.php]: this comes from `SOGoEnableEMailAlarms` in `/etc/e-smith/templates/etc/sogo/sogo.conf/30notifications` although even the sogo documentation is not clear, do you have a suggestion @markVnl

Comment (stephdl) [reply root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Sogo.php]: got it :+1: 


SOGoEnableEMailAlarms | Parameter used to enable email-based alarms on events and tasks.   Defaults to NO when unset.   For this feature to work correctly, one must also set the OCSEMailAlarmsFolderURL parameter and enable the associated cronjob. See the Cronjob — EMail reminders section from this document for more information.
-- | --

Comment (stephdl) [reply root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Sogo.php]: proposition : send email-based alarm on events and tasks

File: root/etc/e-smith/templates/etc/httpd/conf.d/zzz_SOGo.conf/10base (hunk 3 above)
Comment (DavidePrincipi) [reply root/etc/e-smith/templates/etc/httpd/conf.d/zzz_SOGo.conf/10base]: Maybe I'm missing soething but... Isn't the ActiveSync endpoint defined as a "global" alias in Apache? so it can be accessible with any host name!

Comment (markVnl) [reply root/etc/e-smith/templates/etc/httpd/conf.d/zzz_SOGo.conf/10base]: >the documentation stated that multiple vhost is possible, I had two solutions, modify the documentation to say only one vhost is usable, this is the simplest way.

I took this functionally out a few years ago (for reason above), do not have the impression it has much demand.

Doc was/is changed about a month ago and made it to `latest',   
however it does not seem to land in v7

http://docs.nethserver.org/en/latest/sogo.html#access-sogo-on-an-exclusive-hostname

another minor argument to keep the single host is preventing  users wanting to setup a  fully multi tenant configuration. (eg `mail.ltdA.com` and `mail.ltdB.com`) which is not really possible with our accounts provider(s).

>mail.subdomain.com, but for a web application it is probably a non sense, it will probably sogo.subdomain.com and you will fail to configure your mobile activesync settings. It is a real case I got it this weekend after I did a reset to my cell

This makes sense and could be even needed for autodiscover. 
For the latter we could add 'autodiscover.yourdomain.com' in the future.

_in conclusion:_
not a deal breaker for me. IMO we are making it complicated and introduce noise on expectations on the configuration you can achieve.

Comment (markVnl) [reply root/etc/e-smith/templates/etc/httpd/conf.d/zzz_SOGo.conf/10base]: > Maybe I'm missing soething but... Isn't the ActiveSync endpoint defined as a "global" alias in Apache? so it can be accessible with any host name!

IIRC at the moment is not, just a proxy pass, could we a good idea though

File: root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Sogo.php (hunk 2 above)
Comment (markVnl) [reply root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Sogo.php]: none would expect the behavior is to get notified by an email if a task expires.
Will try to find this out..

_if so_ proposition :

<s>`Send email to user for expired events and tasks`</s>

**EDIT**

 unclear when a notification email is send fo 2nd proposition is:

`Send email to user for events and tasks`

**Comment by stephdl** [comment]
Well back to the single vhost for SOGo, OT of this issue, this should be opened by another thread.

Do you agree to squash and merge @DavidePrincipi and @markVnl

**Comment by DavidePrincipi** [comment]
As far as I understand, it's ok for me ;)
